curriculum/challenges/english/25-front-end-development/lecture-working-with-the-dom-click-events-and-web-apis/673368e7bd043f331919514d.md

```diff
@@ -10,6 +10,88 @@ dashedName: how-do-the-navigator-window-and-document-work
 
 Watch the lecture video and answer the questions below.
 
+# --transcript--
+
+How Do the `Navigator`, `Window`, and `Document` Work?
+
+When working with the DOM, you will often come across the `Navigator`, `Window`, and `Document` interfaces. An interface is a collection of methods and properties that define a particular object.
+
+In this lecture, we will explore how these interfaces work and how you can use them in your web applications.
+
+Let's start by looking at the `Navigator` interface.
+
+The `Navigator` interface provides information about the browser environment, such as the user agent string, the platform, and the version of the browser. A user agent string is a text string that identifies the browser and operating system being used.
+
+Here is an example of how to access the user agent string using the `Navigator` interface:
+
+```js
+console.log(navigator.userAgent);
+```
+
+The result will be a string that contains information about the browser and operating system being used.
+
+Here is an example string that you might see:
+
+```md
+Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36
+```
+
+This could be helpful if you want to display different content based on the user's browser or operating system.
+
+Another useful property of the `Navigator` interface is the `language` property, which returns the language of the browser. You can use this property to display content in the user's preferred language.
+
+Here is an example of using the `language` property:
+
+```js
+console.log(navigator.language);
+```
+
+The result will be a string that contains the language code of the browser. If your preferred language is English, it will return `en-US`.
+
+Next, let's look at the `Window` interface.
+
+The `Window` interface represents the browser window that contains the DOM document. It provides methods and properties for interacting with the browser window, such as resizing the window, opening new windows, and navigating to different URLs.
+
+Here is an example of working with the `innerWidth` property of the `Window` interface:
+
+```js
+console.log(window.innerWidth);
+```
+
+The result will be the width of the browser window in pixels. For example, if the browser window is 800 pixels wide, it will return 800.
```

Comment:
```suggestion
The result will be the width of the browser window in pixels. For example, if the browser window is 800 pixels wide, it will return `800`.
```